src/plopp/backends/matplotlib/residuals.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+import matplotlib.pyplot as plt
+
+from ..protocols import FigureLike
+from .utils import copy_figure, make_figure
+
+
+class ResidualPlot:
+    def __init__(self, main_panel: FigureLike, res_panel: FigureLike):
+        self.main_panel = main_panel
+        self.res_panel = res_panel
+        self.panels = [self.main_panel, self.res_panel]
+
+    def __getitem__(self, key):
+        return self.panels[key]
+
+    def __repr__(self):
+        return f"ResidualPlot(main_panel={self.main_panel}, res_panel={self.res_panel})"
+
+    def _repr_mimebundle_(self, *args, **kwargs) -> dict:
+        return self.main_panel._repr_mimebundle_(*args, **kwargs)
+
+
+def residuals(main_fig: FigureLike, reference: FigureLike) -> ResidualPlot:
+    """
+    Create a residual plot from two figures, using the data from the second figure as
+    the reference the residuals are computed from.
+
+    Parameters
+    ----------
+    main_fig:
+        The main figure.
+    reference:
+        The reference figure.
```

Comment:
Missed this earlier, but wouldn't it make more sense to accept a data array instead of a figure? Why would you require the user to plot the reference data first?

Reply:
Yes, I'm starting to think that the whole approach using operators is a bad idea. I also ran into problems if, for example, the user made a figure where they changed the line style or the line color. Then I had to make sure that was copied over.

I think we need to go back to some of the suggestions in #201 and use a dedicated function, and drop the operators.